x-pack/plugins/cases/server/client/attachments/bulk_delete.ts

```diff
@@ -137,9 +157,27 @@ const getFiles = async (
     throw Boom.badRequest(`Failed to delete files because filed ids were invalid: ${invalidIds}`);
   }
 
-  if (validFiles.length <= 0) {
-    throw Boom.badRequest('Failed to find files to delete');
-  }
-
   return validFiles.map((fileInfo) => fileInfo.data);
 };
+
+export const retrieveFilesIgnoringNotFound = (
+  results: Array<PromiseResult<File<unknown>>>,
+  fileIds: BulkDeleteFileArgs['fileIds'],
+  logger: Logger
+) => {
+  const files: File[] = [];
+
+  results.forEach((result, index) => {
+    if (result.isFulfilled) {
+      files.push(result.value);
+    } else if (result.reason instanceof FileNotFoundError) {
+      logger.warn(`Failed to find file id: ${fileIds[index]}: ${result.reason}`);
+    } else if (result.reason instanceof Error) {
```

Comment:
Unfortunately I don't think I can test this particular code path with an end to end test. The issue is that if a user doesn't have cases permission it'll fail before this function when we do the `.resolve`. If the user only has delete permissions, they'll still have read and write access to the file saved objects.

Reply:
No worries. The unit tests you have are enough.